resources/metadata/abapEnvironmentAssembly.yaml

```diff
@@ -0,0 +1,202 @@
+        type: jenkins
+    params:
+      - name: cfApiEndpoint
+        type: string
+        description: Cloud Foundry API endpoint
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        - GENERAL
+        mandatory: false
+        aliases:
+          - name: cloudFoundry/apiEndpoint
+      - name: cfOrg
+        type: string
+        description: CF org
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        - GENERAL
+        mandatory: false
+        aliases:
+          - name: cloudFoundry/org
+      - name: cfSpace
+        type: string
+        description: CF Space
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        - GENERAL
+        mandatory: false
+        aliases:
+          - name: cloudFoundry/space
+      - name: cfServiceInstance
+        type: string
+        description: Parameter of ServiceInstance Name to delete CloudFoundry Service
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        - GENERAL
+        mandatory: false
+        aliases:
+          - name: cloudFoundry/serviceInstance
+      - name: cfServiceKeyName
+        type: string
+        description: Parameter of CloudFoundry Service Key to be created
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        - GENERAL
+        mandatory: false
+        default: SAP_COM_0582
+        aliases:
+          - name: cloudFoundry/serviceKeyName
+      - name: host
+        description: Specifies the host address of the SAP Cloud Platform ABAP Environment system
+        type: string
+        mandatory: false
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+      - name: username
+        type: string
+        description: User or E-Mail for CF
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+        secret: true
+      - name: password
+        type: string
+        description: User Password for CF User
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+        secret: true
+      - name: PackageType
+        type: string
+        description: Type of the delivery package(AOI, CSP, CPK) as provided by AAKaaS 
+        mandatory: true
+        possibleValues:
+          - AOI
+          - CSP
+          - CPK
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: PackageType
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+      - name: PackageName
+        type: string
+        description: Name of delivery package as provided by AAKaaS
+        mandatory: true
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: PackageName
+        scope:
+        - GENERAL
```

Comment:
If a paramter has the scope GENERAL, it can be configured in the GENERAL section of the configuration (if a config.yml file is used). The configuration is then valid for all steps with a parameter with this specific name (and the GENERAL scope). This is useful for e.g. "cfApiEndpoint", as a lot of different steps require this parameter.

It is the same for the other scopes. A parameter which is enabled and configured in the STAGES section of the configuration is valid for all steps within the specific stage.

Currently, I would think that the GENERAL scope is not required for the step specific paramters (packageName etc.) and would remove it. It is easier to add it at a later point in time than to remove it.